Prevent ambiguous input arguments for get_variables_metadata

JamesAnstey opened the pull request:
Addresses https://github.com/CMIP-Data-Request/CMIP7_DReq_Software/issues/148

`get_variables_metadata` allowed the user to specify more than one output file name, to get the metadata in json or csv format. Originally I thought that would be useful but now it seems confusing and unnecessary, so to simplify & make the input args usage consistent with `export_dreq_lists`, this PR makes the output file a single file, which is a positional argument.

JamesAnstey committed:
Remove ability for get_variables_metadata.py to write more than one output file type (json & csv).
Instead user can invoke it twice to get different formats of output file.
This makes the positional arguments the same as export_dreq_lists_json, which is simpler for users.

JamesAnstey committed:
revised --variables_metadata option in export_dreq_lists_json for usage consistency with get_variables_metadata

JamesAnstey committed:
parse input lists separated by comma instead of space in get_variables_metadata

JamesAnstey committed:
parse input lists separated by comma instead of space in export_dreq_lists_json
also added short versions of optional args, e.g. "-e" for "--experiments"

JamesAnstey committed:
removed --version argument from export_dreq_lists_json, because it's no longer needed and also confusing
confusing, because the positional arguments need to be specified even though they're not used
not needed, because the automatic API version warning message says what the installed version is

JamesAnstey committed:
updated tests involving get_variables_metadata (removed '-o', comma-separated argument lists)

JamesAnstey commented:
After discussion in https://github.com/CMIP-Data-Request/CMIP7_DReq_Software/issues/148, also changed `get_variables_metadata` and `export_dreq_lists_json` optional argument lists from space-delimited to comma-delimited

JamesAnstey committed:
updated estimate_dreq_volume -v,-e optional args to use comma-delimited lists

rigoudyg commented:
To use json.loads to get a value with argparse :
`    parser.add_argument("--variable_attributes", type=json.loads,
                        help="Attributes of the variable to add")`

sol1105 reviewed (approved):
Thanks, I think that can be merged, I did not notice any issues.

sol1105 commented:
Maybe one thing:

the `get_variables_metadata` could also add the variable UID (in line ~950), maybe it would be useful:
> 'uid': var.uid,

For  (after?) v1.2.2 this would probably hold the CMIP7 compound name (after yesterday's discussion).

JamesAnstey committed:
added UID from Variables table to get_variables_metadata output json file
updated scripts/examples/variables_v1.2.1.json to be consistent with this change

JamesAnstey commented:
> Maybe one thing:
> 
> the `get_variables_metadata` could also add the variable UID (in line ~950), maybe it would be useful:
> 
> > 'uid': var.uid,
> 
> For (after?) v1.2.2 this would probably hold the CMIP7 compound name (after yesterday's discussion).

Added the UID from the Variables table, and the example file `scripts/examples/variables_v1.2.1.json` is updated to include this change.
